lib/search/query_components/query.rb

```diff
@@ -3,16 +3,27 @@
 
 module QueryComponents
   class Query < BaseComponent
+    def initialize(content_index_names:, metasearch_index:, search_params: Search::QueryParameters.new)
+      @content_index_names = content_index_names
+      @metasearch_index = metasearch_index
+
+      super(search_params)
+    end
```

Comment:
meaning on `content_index_names` is a special case (if you move the rest to the `BaseComponent`